tests/local/network.go

```diff
@@ -453,74 +434,146 @@ func (n *LocalNetwork) RelayMessage(ctx context.Context,
 }
 
 func (n *LocalNetwork) setAllSubnetValues() {
-	subnetIDs := n.manager.GetSubnets()
+	subnetIDs := n.GetSubnetsInfo()
 	Expect(len(subnetIDs)).Should(Equal(2))
 
-	n.subnetAID = subnetIDs[0]
-	n.setSubnetValues(n.subnetAID)
-
-	n.subnetBID = subnetIDs[1]
-	n.setSubnetValues(n.subnetBID)
+	n.setSubnetValues(n.tmpnet.GetSubnet("A").SubnetID)
+	n.setSubnetValues(n.tmpnet.GetSubnet("B").SubnetID)
 
 	n.setPrimaryNetworkValues()
 }
 
 func (n *LocalNetwork) TearDownNetwork() {
 	log.Info("Tearing down network")
-	Expect(n.manager).ShouldNot(BeNil())
-	Expect(n.manager.TeardownNetwork()).Should(BeNil())
+	Expect(n).ShouldNot(BeNil())
+	Expect(n.tmpnet).ShouldNot(BeNil())
+	Expect(n.tmpnet.Stop(context.Background())).Should(BeNil())
 	Expect(os.Remove(n.warpChainConfigPath)).Should(BeNil())
 }
 
-func (n *LocalNetwork) AddSubnetValidators(ctx context.Context, subnetID ids.ID, nodeNames []string) {
-	_, err := n.anrClient.AddSubnetValidators(ctx, []*rpcpb.SubnetValidatorsSpec{
-		{
-			SubnetId:  subnetID.String(),
-			NodeNames: nodeNames,
-		},
-	})
+func (n *LocalNetwork) AddSubnetValidators(ctx context.Context, subnetID ids.ID, count uint) {
+	Expect(count > 0).Should(BeTrue(), "can't add 0 validators")
+	Expect(uint(len(n.extraNodes)) >= count).Should(
+		BeTrue(),
+		"not enough extra nodes to use",
+	)
+
+	subnet := n.tmpnet.Subnets[slices.IndexFunc(
+		n.tmpnet.Subnets,
+		func(s *tmpnet.Subnet) bool { return s.SubnetID == subnetID },
+	)]
+
+	// consume some of the extraNodes
+	newValidatorNodes := n.extraNodes[0:count]
+	slices.Delete(n.extraNodes, 0, int(count))
+
+	apiURI, err := n.tmpnet.GetURIForNodeID(subnet.ValidatorIDs[0])
 	Expect(err).Should(BeNil())
 
-	// Add the new node names
-	n.subnetNodeNames[subnetID] = append(n.subnetNodeNames[subnetID], nodeNames...)
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
```

Comment:
is 30 seconds longs enough for adding the validators and restarting the nodes on any machine?

Reply:
it's a good question, but I'm not sure of the answer. all I can say is that it's good enough for my azure dev env, and for CI. (I've never seen this timeout occur, either locally or in CI.) if I remember correctly, the call that uses this context refuses to run if there isn't _some_ timeout (or deadline) specified. I'm happy to bump this up to whatever you think is reasonable.

Reply:
Yeah, let's bump it that should be too long to fail if things are working, but not too long that we're waiting around forever if something is broken. 60-120 seconds should be fine.

Reply:
addressed in https://github.com/ava-labs/teleporter/pull/401/commits/6ede518c697c3315170b18dc3f45e66bea3787d8